### ISSUE TITLE
Retry policy: panic: attempt to shift left with overflow

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#421]
I started sozu with the basic configuration minus those changes:
- trace log level
- 1 worker
- worker_automatic_restart = false

And got that panic: 
```
thread 'main' panicked at 'attempt to shift left with overflow', lib/src/network/retry.rs:65:36
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::print
             at /checkout/src/libstd/sys_common/backtrace.rs:68
             at /checkout/src/libstd/sys_common/backtrace.rs:57
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/panicking.rs:381
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:397
   4: sozu::register_panic_hook::{{closure}}
             at bin/src/main.rs:263
   5: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:578
   6: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:538
   7: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:522
   8: rust_begin_unwind
             at /checkout/src/libstd/panicking.rs:498
   9: core::panicking::panic_fmt
             at /checkout/src/libcore/panicking.rs:71
  10: core::panicking::panic
             at /checkout/src/libcore/panicking.rs:51
  11: <sozu_lib::network::retry::ExponentialBackoffPolicy as sozu_lib::network::retry::RetryPolicy>::fail
             at lib/src/network/retry.rs:65
  12: <sozu_lib::network::retry::RetryPolicyWrapper as sozu_lib::network::retry::RetryPolicy>::fail
             at lib/src/network/retry.rs:124
  13: <sozu_lib::network::http::ServerConfiguration as sozu_lib::network::ProxyConfiguration<sozu_lib::network::http::Client>>::connect_to_backend::{{closure}}
             at lib/src/network/http.rs:736
  14: <core::option::Option<T>>::map
             at /checkout/src/libcore/option.rs:404
  15: <sozu_lib::network::http::ServerConfiguration as sozu_lib::network::ProxyConfiguration<sozu_lib::network::http::Client>>::connect_to_backend
             at lib/src/network/http.rs:730
  16: sozu_lib::network::proxy::Server::connect_to_backend
             at lib/src/network/proxy.rs:718
  17: sozu_lib::network::proxy::Server::interpret_client_order
             at lib/src/network/proxy.rs:796
  18: sozu_lib::network::proxy::Server::ready
             at lib/src/network/proxy.rs:851
  19: sozu_lib::network::proxy::Server::run
             at lib/src/network/proxy.rs:368
  20: sozu::worker::begin_worker_process
             at bin/src/worker.rs:86
  21: sozu::main
             at bin/src/main.rs:98
  22: std::rt::lang_start::{{closure}}
             at /checkout/src/libstd/rt.rs:74
  23: std::panicking::try::do_call
             at /checkout/src/libstd/rt.rs:59
             at /checkout/src/libstd/panicking.rs:480
  24: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:101
  25: std::rt::lang_start_internal
             at /checkout/src/libstd/panicking.rs:459
             at /checkout/src/libstd/panic.rs:365
             at /checkout/src/libstd/rt.rs:58
  26: std::rt::lang_start
             at /checkout/src/libstd/rt.rs:74
  27: main
  28: __libc_start_main
  29: _start
```

For it to panic, I had to send *a lot* of requests to an app that has no available backends (the default app).

To easily reproduce, carefully follow those steps:
- start sozu and redirect its output to `/dev/null`: `RUST_BACKTRACE=1 ../target/debug/sozu start -c config.toml > /dev/null`
- create this script:
```
#!/usr/bin/env bash

while true;
  do curl lolcatho.st:8080 -s &
done
```
this script does a lot of requests without waiting for the previous one to finish (it puts the `curl` in the background).

- start the script with: `./script.sh 2>&1 >/dev/null`

It should panic in a few seconds

On my machine, I had to redirect the output to `/dev/null` because otherwise logs printing would slow down sozu and my tmux in general and the panic wouldn't trigger. This may not be necessary for you.

**Comment by Geal** [comment]
This should be fixed by 62fc047 and eb1d606

**Comment by BlackYoup** [comment]
I couldn't reproduce it since those commits
